automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -244,6 +244,95 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
+              withMethod("reader-cannot-import-method", MethodData.SimpleMethod) { methodName =>
+                val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+                // try to import a method config from another workspace
+                val sourceMethodConfig = Map(
+                  "name" -> method.methodName,
+                  "namespace" -> method.methodNamespace,
+                  "workspaceName" -> Map(
+                    "namespace" -> projectName,
+                    "name" -> sourceWorkspaceName))
+
+                val destMethodConfig = Map(
+                  "name" -> s"destination-${method.methodName}",
+                  "namespace" -> s"destination-${method.methodNamespace}",
+                  "workspaceName" -> Map(
+                    "namespace" -> projectName,
+                    "name" -> destWorkspaceName)
+                )
+                Rawls.methodConfigs.createMethodConfigInWorkspace(
+                  projectName, sourceWorkspaceName, method, method.methodNamespace, method.methodName, 1,
+                  Map.empty, Map.empty, method.rootEntityType)(ownerAuthToken)
+
+                eventually {
+                  val copyFromWorkspaceException = intercept[RestException] {
+                    Rawls.methodConfigs.copyMethodConfigFromWorkspace(sourceMethodConfig, destMethodConfig)(studentAToken)
+                  }
+                  assertExceptionStatusCode(copyFromWorkspaceException, 403)
+                }
+
+                // try to import a method config from the method repo
+                val methodRepoConfig = Map(
+                  "methodRepoNamespace" -> SimpleMethodConfig.configNamespace,
+                  "methodRepoName" -> SimpleMethodConfig.configName,
+                  "methodRepoSnapshotId" -> SimpleMethodConfig.snapshotId,
+                  "destination" -> destMethodConfig
+                )
+
+                // studentA needs permission to access the method config or importing from method repo will return 404 not 403
+                Orchestration.methodConfigurations.setMethodConfigPermission(
+                  SimpleMethodConfig.configNamespace,
+                  SimpleMethodConfig.configName,
+                  SimpleMethodConfig.snapshotId,
+                  studentA.email,
+                  "OWNER"
+                )(ownerAuthToken)
+
+                eventually {
+                  val copyFromMethodRepoException = intercept[RestException] {
+                    Rawls.methodConfigs.copyMethodConfigFromMethodRepo(methodRepoConfig)(studentAToken)
+                  }
+                  assertExceptionStatusCode(copyFromMethodRepoException, 403)
+                }
+              }(ownerAuthToken)
+            }(ownerAuthToken)
+          }(ownerAuthToken)
+        }
```

Comment:
this feels a bit like 2 different tests - one for copying the mc from a workspace and importing the mc from the method repo.

Reply:
agreed. I left it as one test so it'd be as close to the original FC-UI test as possible, but I think it makes more sense as two